test/integration/framework/etcd.go

```diff
@@ -128,7 +129,15 @@ func RunCustomEtcd(dataDir string, customFlags []string) (url string, stopFn fun
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 	stop := func() {
-		cancel()
+		// try to exit etcd gracefully
+		cmd.Process.Signal(syscall.SIGTERM)
```

Comment:
Since this is a test, should we also handle SIGINT?

Reply:
The command Context sends Sigkill when the Context is cancelled

Reply:
Previously, we sent Sigkill, when doing the 'cancel'

Reply:
The change seems good for me because https://pkg.go.dev/os/exec#CommandContext shows
```
The provided context is used to kill the process (by calling os.Process.Kill) if the context becomes done before the command completes on its own. 
```

/test pull-kubernetes-e2e-kind-ipv6
/triage accepted